packages/@aws-cdk/aws-ecs-patterns/lib/ecs/queue-processing-ecs-service.ts

```diff
@@ -100,17 +101,30 @@ export class QueueProcessingEc2Service extends QueueProcessingServiceBase {
 
     // Create an ECS service with the previously defined Task Definition and configure
     // autoscaling based on cpu utilization and number of messages visible in the SQS queue.
-    this.service = new Ec2Service(this, 'QueueProcessingService', {
-      cluster: this.cluster,
-      desiredCount: this.desiredCount,
-      taskDefinition: this.taskDefinition,
-      serviceName: props.serviceName,
-      minHealthyPercent: props.minHealthyPercent,
-      maxHealthyPercent: props.maxHealthyPercent,
-      propagateTags: props.propagateTags,
-      enableECSManagedTags: props.enableECSManagedTags,
-      deploymentController: props.deploymentController,
-    });
+    if (!this.node.tryGetContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT)) {
+      this.service = new Ec2Service(this, 'QueueProcessingService', {
+        cluster: this.cluster,
+        desiredCount: this.desiredCount,
+        taskDefinition: this.taskDefinition,
+        serviceName: props.serviceName,
+        minHealthyPercent: props.minHealthyPercent,
+        maxHealthyPercent: props.maxHealthyPercent,
+        propagateTags: props.propagateTags,
+        enableECSManagedTags: props.enableECSManagedTags,
+        deploymentController: props.deploymentController,
+      });
+    } else {
+      this.service = new Ec2Service(this, 'QueueProcessingService', {
+        cluster: this.cluster,
+        taskDefinition: this.taskDefinition,
+        serviceName: props.serviceName,
+        minHealthyPercent: props.minHealthyPercent,
+        maxHealthyPercent: props.maxHealthyPercent,
+        propagateTags: props.propagateTags,
+        enableECSManagedTags: props.enableECSManagedTags,
+        deploymentController: props.deploymentController,
+      });
+    }
```

Comment:
```suggestion
        const desiredCount = this.node.tryGetContext(cxapi.ECS_REMOVE_DEFAULT_DESIRED_COUNT) ? undefined : this.desiredCount;
    this.service = new Ec2Service(this, 'QueueProcessingService', {
      cluster: this.cluster,
      desiredCount: desiredCount,
      taskDefinition: this.taskDefinition,
      serviceName: props.serviceName,
      minHealthyPercent: props.minHealthyPercent,
      maxHealthyPercent: props.maxHealthyPercent,
      propagateTags: props.propagateTags,
      enableECSManagedTags: props.enableECSManagedTags,
      deploymentController: props.deploymentController,
    });
```

Reply:
I was trying to separate out the logic so that when the time comes for the next major release, it's just easier to remove the feature flag and clean up since all you need to do is delete the if block. Let me know if you disagree

Reply:
I think it's more confusing this way honestly. When feature flag is removed, you just have to delete two lines with my suggestion.

Reply:
I agree with @SoManyHs here, I think the repetition is more confusing in this case.